### PR TITLE
Add and use get_module_version for cache tokens

### PR DESCRIPTION
**Pull request by yarikoptic**
Ran into crash of dandi-cli tests on our docker `dandiarchive/nwb-healthstatus:pynwb-1.2.0` with some arbitrarily old hdmf before it got `__version__`.  I do not think we should puke like that so decided to resort to asking pkg_resources if no `__version__`

Learning typing while at it -- @jwodder, how to properly annotate for a Module?  please suggest a fix

### CODE REVIEW

File: dandi/utils.py [hunk 1]
```diff
@@ -686,3 +688,30 @@ def is_url(s):
     TODO: redo
     """
     return s.lower().startswith(("http://", "https://", "dandi://"))
+
+
+def get_module_version(module: Union[str, "Module"]) -> object:
```
Comment (jwodder) [inline dandi/utils.py:693]: `"Module"` should be `types.ModuleType`, and the return value should be `Any`.

File: dandi/utils.py [hunk 2]
```diff
@@ -686,3 +688,30 @@ def is_url(s):
...
+        module = sys.modules.get(module, None)
+    else:
+        mod_name = module.__name__.split(".", 1)[0]
+
+    version = getattr(module, "__version__", None)
+    if version is None:
+        # Let's use the standard Python mechanism if underlying module
+        # did not provide __version__
+        try:
+            version = pkg_resources.get_distribution(mod_name).version
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply dandi/utils.py]: done

File: dandi/utils.py [hunk 3]
```diff
@@ -686,3 +688,30 @@ def is_url(s):
...
+    Return module's `__version__` and if present, or use pkg_resources
+    to get version.
+
+    Returns
+    -------
+    object
+    """
+    if isinstance(module, str):
+        mod_name = module
+        module = sys.modules.get(module, None)
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply dandi/utils.py]: done

File: dandi/utils.py (hunk 1 above)
Comment (yarikoptic) [reply dandi/utils.py]: done -- made it into
```
def get_module_version(module: Union[str, types.ModuleType]) -> Optional[str]:
```
assuming that those modules we deal with would have it as a `str`

File: dandi/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yarikoptic) [reply dandi/utils.py]: hm, I have added 
```
setup.cfg:    importlib-metadata;  python_version < "3.8"

```
to `install_requires` but it seems (CI is rerunning now after a fresh commit; prior run was https://github.com/dandi/dandi-cli/runs/2300820706?check_suite_focus=true) that python 3.7 did not get it installed -- what is the catch @jwodder ?

[the remark this answers is not in the document]
Comment (jwodder) [reply dandi/utils.py]: When installing from PyPI, the package must be imported as `importlib_metadata`, not `importlib.metadata`.  To cover both cases, write something like this:

```
try:
    from importlib.metadata import version
except ImportError:
    from importlib_metadata import version
```

[the remark this answers is not in the document]
Comment (yarikoptic) [reply dandi/utils.py]: oh, ok -- will do

**Comment by yarikoptic** [comment]
push fails are about `most_recent_version` ... addressed in master, let's proceed

**Pull request merged**